lib/logstash/inputs/udp.rb

```diff
@@ -114,4 +128,21 @@ def stop
     @udp.close rescue nil
   end
 
+  def identity(client)
+    [client[3], client[1]].compact.join("-")
+  end
+
+  def need_identity_map?
+    case @codec
+    when LogStash::Codecs::Netflow
+      true
+    when LogStash::Codecs::IdentityMapCodec
+      true
+    when LogStash::Codecs::Multiline
+      true
+    else
+      false
+    end
+  end
+
```

Comment:
We cannot use the LogStash::Codecs::IdentityMapCodec directly, we can simplify the predicate with
```ruby
@codec.class == LogStash::Codecs::Netflow || @codec.class == LogStash::Codecs::Multiline
```

@guyboertje Do you remember why we decided adding a method to the codec base class to make it explicit that a codec support or could benefit of using the `identity_map`?